File: apps/scanner/lib/product_label.dart
/// What a product is called in full: "Shure SM58", or just "Kaltgerätekabel"
/// for one nobody makes in particular, which the server sends without a
/// manufacturer. The web joins them the same way — `apps/web/src/lib/product-label.ts`.
String productLabel(String? manufacturerName, String name) =>
    manufacturerName == null ? name : '$manufacturerName $name';

File: apps/web/src/lib/product-label.ts
/**
 * What a product is called wherever it is named in full: "Shure SM58", or just
 * "Kaltgerätekabel" for a product nobody makes in particular. The one place the
 * maker and the name are joined, so a missing maker never leaves a gap or a
 * placeholder on a delivery note.
 */
export function productLabel(product: {
	name: string;
	manufacturer: { name: string } | null;
}): string {
	return makerAndName(product.manufacturer?.name, product.name);
}

/** `productLabel` for rows that carry the maker as a flat `manufacturerName`. */
export function makerAndName(manufacturerName: string | null | undefined, name: string): string {
	return manufacturerName ? `${manufacturerName} ${name}` : name;
}
